### ISSUE TITLE
Steam doesn't redraw in tiling window manager

### ISSUE THREAD

**Issue by Thorinori** [ValveSoftware/steam-for-linux#10806]
#### Your system information

* Steam client version (build number or date): 1713317598
* Distribution (e.g. Ubuntu): EndeavourOS (Arch based)
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/15049951/steam-logs.tar.gz)
* GPU: AMD Radeon RX 6800 XT

#### Please describe your issue in as much detail as possible:
I use i3 as my window manager, and when I change workspaces on the same monitor Steam is running in, or a game takes focus in the same workspace, Steam itself no longer redraws until I force the window to move via keyboard commands (it can have focus and not redraw, I specifically need to move it to force a redraw). This has been happening for the last several versions, I am unsure of which version it was introduced in but it only started in the last 2 to 3 weeks at most.

#### Steps for reproducing this issue:
1) Have i3 (or potentially other tiling wm)
2) Run Steam
3) Go to another workspace and open something on same monitor, or open a game in same workspace
4) Come back and/or close game, Steam shouldn't be rerdrawing automatically anymore.

**Comment by nuggs** [comment]
I've noticed the same issue lately running stock arch linux.

I just wanted to put it here for anyone else that stumbles across this, if you resize either directly or by resizing something else on the same workspace it will be responsive again until you switch workspaces.

**Comment by sinsinewave** [comment]
can confirm as well, steam doesn't redraw unless resized on i3

i'm on i3 on Void Linux with an RX 6800 XT as well; and noticed that everyone here has a (relatively modern) AMD GPU. now wondering if anyone has had this on NVidia or Intel, or if this is AMD-specific?

**Comment by michael-n0813** [comment]
Same issue on arch with awesomeWM tiling windows manager. They released this mess into stable so now everyone gets to play.

**Comment by Philanche** [comment]
Can confirm, I'm on arch using i3wm, I get the same issue on my laptop when using the AMD iGPU, which is the default, however when I force steam to run through my NVidia dedicated GPU, I can switch back and forth and no issue appears, so it seems to be an AMD only issue, or at the very least, not affect NVidia GPUs.

**Comment by lun-4** [comment]
I'm also reproducing this on Void Linux glibc, awesomewm (however, not in any tiling mode), steam version 1716242052, and AMD RX 580 2048SP (radeonsi, polaris10)

**Comment by jpgsmits** [comment]
Same issue here using Arch Linux (kernel 6.8.9-arch1-2), using AwesomeWM, running steam version 1716242052 on an AMD RX 7900 XTX. Workaround for me is to minimize and maximize steam and then it redraws again.

**Comment by Arszilla** [comment]
Experiencing the same on 6.8.11-amd64 kernel on Arch using i3. Switching between Store and Library or having "Caching Vulkan Shaders" etc does not update etc. unless I toggle floating and tiling (or vice-versa) for Steam.

**Comment by szmergiell** [comment]
I'm experiencing similar problem with arch, i3 and steam.

I don't fully understand why or claim that it is a "proper" solution, but running steam with `-system-composer` resolved the issue  on my end.

**Comment by clussy-juice** [comment]
Same issue on awesomewm. Kernel 6.9.6, mesa git with radv, steam client beta.

**Comment by jajm** [comment]
Same issue for me but only when "GPU accelerated rendering in web views" is enabled

OpenGL renderer string: AMD Radeon RX 7800 XT (radeonsi, navi32, LLVM 17.0.6, DRM 3.57, 6.9.9-amd64)
i3 version 4.23

As a workaround I run a script that forces Steam window to resize by adding a border and then removing it (https://gist.github.com/jajm/0941d641b92ffb713930ccb43e52359c)

**Comment by ruinedraziel** [comment]
Issues seems very similar, maybe all the same issues?

https://github.com/ValveSoftware/steam-for-linux/issues/6660
https://github.com/ValveSoftware/steam-for-linux/issues/10869

Very, really very, similar behavior.

**Comment by VanackSabbadium** [comment]
I confirm this issue on my system, which is Debian Sid + AwesomeWM.
I'm using Steam in a floating window desktop, but the same issue happens selecting another type of tiling setting.

Resizing or minimizing/maximizing Steam window redraws it and solve the issue temporarily, until another desktop switching.

**Comment by VanackSabbadium** [comment]
> Disabling "Enable GPU accelerated rendering in web views" in interface setting, work for me

I can confirm, it works.

**Comment by Thorinori** [comment]
> Disabling "Enable GPU accelerated rendering in web views" in interface setting, work for me

Can also confirm this works, good find!

**Comment by ruinedraziel** [comment]
> Disabling "Enable GPU accelerated rendering in web views" in interface setting, work for me

Also works for me.

**Comment by Swivelgames** [comment]
Just closed my duplicate issue with instruction to come here instead, and attempt to disable GPU accelerated web views :+1: 

- #11056

**Comment by matoro** [comment]
This is still a problem on i3.  Disabling GPU acceleration indeed fixes the issue, but it causes big picture mode performance to be crippled into slideshow territory.

One workaround I found was fullscreen+un-fullscreen (press super-F twice) will force it to redraw.  Does anybody know of any automatic workarounds that might be able to be done solely via i3 config?

**Comment by ruinedraziel** [comment]
@matoro 
Yes, can't find the original comment right away, so the credit is not mine. Thx for the guy who posted this btw.

I created launcher for steam on ~/.local/share/applications, regular copy from the steam commom launcher, then added the paramenter -system-composer to the Exec= line. like this:
Exec=/usr/bin/steam -system-composer %U

Can run fine with GPU acceleration, no issues, even reinstalled the system, since is on .local the config remained as part of my home folder. Pretty elegant.

Can you see if this also helps you?

**Comment by matoro** [comment]
> Replying to https://github.com/ValveSoftware/steam-for-linux/issues/10806#issuecomment-3710680425

Wow, this is exactly what I was looking for!  I don't know what `-system-composer` does, but it completely fixes the problem!  Props!!
